src/Sulu/Bundle/MediaBundle/Controller/MediaStreamController.php

```diff
@@ -116,13 +119,35 @@ protected function getFileResponse(
         $fileSize = $fileVersion->getSize();
         $storageOptions = $fileVersion->getStorageOptions();
         $mimeType = $fileVersion->getMimeType();
-        $version = $fileVersion->getVersion();
+        $storageName = $fileVersion->getStorageName();
+        $pathInfo = pathinfo($fileName);
 
-        $path = $this->getStorage()->load($fileName, $version, $storageOptions);
+        $downloadUrl = $this->getStorageManager()->getDownloadUrl($storageOptions, $storageName);
 
-        $response = new BinaryFileResponse($path);
+        // If external download Url exists redirect to it.
+        if (!empty($downloadUrl)) {
+            return new RedirectResponse($downloadUrl);
+        }
 
-        $pathInfo = pathinfo($fileName);
+        $resourcePath = $this->getStorageManager()->load($storageOptions, $storageName);
+
+        if (is_resource($resourcePath)) {
+            $handle = $resourcePath;
```

Comment:
i have now take a deeper look into and the `BinaryFileResponse` uses `X-Sendfile-Type` so if we can we should use this. But for the resources from flysystem it would be cool to implement a `BinaryResourceResponse` which can extract a lot of functionality from the `BinaryFileResponse` and should be contributed to symfony.